frontend/hub/api/request.ts

```diff
@@ -0,0 +1,137 @@
+import { AnsibleError } from '../../common/crud/ansible-error';
+import { getCookie } from '../../common/crud/cookie';
+import { Delay } from '../../common/crud/delay';
+import { TaskResponse } from '../tasks/Task';
+
+interface Options {
+  method: string;
+  headers?: HeadersInit;
+  body?: BodyInit;
+  signal?: AbortSignal;
+}
+
+interface Response<T> {
+  statusCode: number;
+  response: T;
+}
+
+async function hubRequestCommon<T extends object | TaskResponse>(
```

Comment:
I'd recommend adding JSDoc to this function to describe why it was needed (the 202 specific use case).